ccaas_builder/cmd/build/main.go

```diff
@@ -0,0 +1,194 @@
+			return fmt.Errorf("failed to copy build META-INF folder: %s", err)
+		}
+	}
+
+	// Process and update the connections file
+	fileInfo, err := os.Stat(connectionSrcFile)
+	if err != nil {
+		return errors.WithMessagef(err, "%s not found ", connectionSrcFile)
+	}
+
+	connectionFileContents, err := ioutil.ReadFile(connectionSrcFile)
+	if err != nil {
+		return err
+	}
+
+	// read the connection.json file into structure to process
+	var connectionData connection
+	if err := json.Unmarshal(connectionFileContents, &connectionData); err != nil {
+		return err
+	}
+
+	// Treat each of the string fields in the connection.json as Go template
+	// strings. They can be fixed strings, but if they are templates
+	// then the JSON string that is defined in CHAINCODE_AS_A_SERVICE_BUILDER_CONFIG
+	// is used as the 'context' to parse the string
+
+	updatedConnection := connection{}
+	var cfg map[string]interface{}
+
+	cfgString := os.Getenv("CHAINCODE_AS_A_SERVICE_BUILDER_CONFIG")
+	if cfgString != "" {
+		if err := json.Unmarshal([]byte(cfgString), &cfg); err != nil {
+			return fmt.Errorf("Failed to unmarshal %s", err)
+		}
+	}
+
+	updatedConnection.Address, err = execTempl(cfg, connectionData.Address)
+	if err != nil {
+		return fmt.Errorf("Failed to parse the Address field template: %s", err)
+	}
+
+	updatedConnection.DialTimeout, err = execTempl(cfg, connectionData.DialTimeout)
+	if err != nil {
+		return fmt.Errorf("Failed to parse the DialTimeout field template: %s", err)
+	}
+
+	// if connection is TLS Enabled, updated with the correct information
+	// no other information is needed for the no-TLS case, so the default can be assumed
+	// to be good
+	if connectionData.TLS {
+		updatedConnection.TLS = true
+		updatedConnection.ClientAuth = connectionData.ClientAuth
+
+		updatedConnection.RootCert, err = execTempl(cfg, connectionData.RootCert)
+		if err != nil {
+			return fmt.Errorf("Failed to parse the RootCert field template: %s", err)
+		}
+		updatedConnection.ClientKey, err = execTempl(cfg, connectionData.ClientKey)
+		if err != nil {
+			return fmt.Errorf("Failed to parse the ClientKey field template: %s", err)
+		}
+		updatedConnection.ClientCert, err = execTempl(cfg, connectionData.ClientCert)
+		if err != nil {
+			return fmt.Errorf("Failed to parse the ClientCert field template: %s", err)
+		}
+	}
+
+	updatedConnectionBytes, err := json.Marshal(updatedConnection)
+	if err != nil {
+		return fmt.Errorf("failed to marshal updated connection.json file: %s", err)
+	}
+
+	err = ioutil.WriteFile(connectionDestFile, updatedConnectionBytes, fileInfo.Mode())
+	if err != nil {
+		return err
+	}
+
+	return nil
+
+}
+
+// execTempl is a helper function to process a template against a string, and return a string
+func execTempl(cfg map[string]interface{}, inputStr string) (string, error) {
+
+	t, err := template.New("").Option("missingkey=error").Parse(inputStr)
+	if err != nil {
+		fmt.Printf("Failed to parse the template: %s", err)
```

Comment:
Shouldn't errors be written to stderr?

Reply:
good point.. yes they should

Reply:
@andrew-coleman just went and checked - the `logger` that is created at the top of the file is set to use stderr - so all the messages will go to stdout.

Recall now that the peer only captures stderr for the external builders.